components/tls/pkg/tls/util.go

```diff
@@ -50,3 +55,11 @@ func GetSecurityProtocolFromEnv(prefix string) string {
 	}
 	return val
 }
+
+func GetSASLMechanismFromEnv(prefix string) string {
+	val, ok := os.LookupEnv(fmt.Sprintf("%s%s", prefix, EnvSASLMechanismSuffix))
+	if !ok {
```

Comment:
:thought_balloon: We should probably check this is a supported mechanism we know how to handle

Reply:
Whatever we retrieve here; see my other comment for further clarification

Reply:
This is currently not doing anymore than `GetSecurityProtocolFromEnv` function just next to it. 
We can include in further re-factor but for now I just have seen this one as simple helper to get the mechanism and logic for validation to be in place that we use it.